--- origsrc/md5-1.1.1/Makefile	2007-10-10 19:02:56.000000000 -0500
+++ src/md5-1.1.1/Makefile	2008-02-13 21:39:53.437500000 -0600
@@ -10,25 +10,25 @@
 
 MD5_OBJS= src/md5.o src/md5lib.o $(COMPAT_O)
 MD5_LUAS= src/md5.lua
-MD5_LIBNAME = core.so
+MD5_LIBNAME = core.dll
 
 DES56_OBJS= src/des56.o src/ldes56.o
-DES56_LIBNAME= des56.so
+DES56_LIBNAME= des56.dll
 
 all: src/$(MD5_LIBNAME) src/$(DES56_LIBNAME)
 
 src/$(MD5_LIBNAME) : $(MD5_OBJS)
-	export MACOSX_DEPLOYMENT_TARGET="10.3"; $(CC) $(CFLAGS) $(LIB_OPTION) -o src/$(MD5_LIBNAME) $(MD5_OBJS)
+	export MACOSX_DEPLOYMENT_TARGET="10.3"; $(CC) $(CFLAGS) $(LIB_OPTION) -o src/$(MD5_LIBNAME) $(MD5_OBJS) $(LIBS)
 
 src/$(DES56_LIBNAME) : $(DES56_OBJS)
-	export MACOSX_DEPLOYMENT_TARGET="10.3"; $(CC) $(CFLAGS) $(LIB_OPTION) -o src/$(DES56_LIBNAME) $(DES56_OBJS)
+	export MACOSX_DEPLOYMENT_TARGET="10.3"; $(CC) $(CFLAGS) $(LIB_OPTION) -o src/$(DES56_LIBNAME) $(DES56_OBJS) $(LIBS)
 
 $(COMPAT_DIR)/compat-5.1.o: $(COMPAT_DIR)/compat-5.1.c
 	$(CC) -c $(CFLAGS) -o $@ $(COMPAT_DIR)/compat-5.1.c
 
 install: src/$(MD5_LIBNAME) src/$(DES56_LIBNAME)
 	mkdir -p $(LUA_LIBDIR)/md5
-	cp src/$(MD5_LIBNAME) $(LUA_LIBDIR)/md5/core.so
+	cp src/$(MD5_LIBNAME) $(LUA_LIBDIR)/md5/
 	mkdir -p $(LUA_DIR)
 	cp $(MD5_LUAS) $(LUA_DIR)
 	cp src/$(DES56_LIBNAME) $(LUA_LIBDIR)
--- origsrc/md5-1.1.1/config	2007-10-29 18:00:23.000000000 -0500
+++ src/md5-1.1.1/config	2008-02-13 21:36:30.609375000 -0600
@@ -24,7 +24,7 @@
 COMPAT_DIR= ../compat/src
 
 # Compilation directives
-WARN= -O2 -Wall -fPIC -W -Waggregate-return -Wcast-align -Wmissing-prototypes -Wnested-externs -Wshadow -Wwrite-strings -Wpointer-arith
+WARN= -O2 -Wall -DPIC -W -Waggregate-return -Wcast-align -Wmissing-prototypes -Wnested-externs -Wshadow -Wwrite-strings -Wpointer-arith
 INCS= -I$(LUA_INC) -I$(COMPAT_DIR)
 CFLAGS= $(WARN) $(INCS)
 CC= gcc
